src/Amqp/Akka.Streams.Amqp.RabbitMq/AmqpConnector.cs

```diff
@@ -66,10 +70,14 @@ public static IConnection NewConnection(IConnectionFactory factory, IAmqpConnect
                 case AmqpConnectionDetails details:
                 {
                     if (details.HostAndPortList.Count == 0)
-                        throw new ArgumentException("You need to supply at least one host/port pair.", nameof(settings));
+                        throw new ArgumentException("You need to supply at least one host/port pair.",
+                            nameof(settings));
 
                     return factory.CreateConnection(details.HostAndPortList
-                        .Select(pair => new AmqpTcpEndpoint(pair.host, pair.port, details.Ssl)).ToList());
+                        .Select(pair =>
+                            details.Ssl == null
+                                ? new AmqpTcpEndpoint(pair.host, pair.port)
```

Comment:
Used a ternary operator to determine if we have SSL settings or not - use the appropriate CTOR depending.